db/warehouse/migrate/20240514131401_require_client_alert_expiry.rb

```diff
@@ -0,0 +1,17 @@
+#  Copyright 2016 - 2024 Green River Data Analysis, LLC
+#
+#  License detail: https://github.com/greenriver/hmis-warehouse/blob/production/LICENSE.md
+#
+
+class RequireClientAlertExpiry < ActiveRecord::Migration[7.0]
+  def up
+    unexpired = Hmis::ClientAlert.with_deleted.where(expiration_date: nil)
+    unexpired.update_all("expiration_date=created_at + interval '1 month'")
+
+    safety_assured { change_column_null :hmis_client_alerts, :expiration_date, false }
```

Comment:
I wanted to double check about this `safety_assured` -- I know that's something we have done in the past, but would also be happy to implement it as suggested by strong migrations (add a check constraint and validate it in a separate migration). Would those migrations need to be in two different releases?

Reply:
Hmm. This migration would be fine, but I think this database change is a little too high-commitment, actually.

If down the line we have another installation that _doesn't_ want to use Expiration dates on alerts at all, we will want to accommodate that alongside this by using a different client_alert form (i.e. changing `"required": true`). (I'm not suggesting making that configurability change now, but we may want to make it later. I know right now the form is `static`).

Does that make sense? I think as a lower-commitment change, we can just manually validate the presence of the expiration date in `CreateClientAlert`. Then it'll be easier later on to make the expiration date required-ness configurable.

Reply:
Thanks, good point - I kind of wondered about this, should have brought it up before implementation! 🤦